pytorch_lightning/utilities/apply_func.py

```diff
@@ -23,9 +23,10 @@
 import numpy as np
 import torch
 
-from pytorch_lightning.utilities.imports import _compare_version, _TORCHTEXT_AVAILABLE
+from pytorch_lightning.utilities.imports import _compare_version, _TORCHTEXT_LEGACY
+from pytorch_lightning.utilities.warnings import rank_zero_warn
 
-if _TORCHTEXT_AVAILABLE:
+if _TORCHTEXT_LEGACY:
     if _compare_version("torchtext", operator.ge, "0.9.0"):
         from torchtext.legacy.data import Batch
     else:
```

Comment:
The

```python
else:
   Batch = type(None)
```

below can be removed right? we only access it if `_TORCHTEXT_LEGACY is True`

Reply:
Yeah we could potentially remove this and add conditional imports in the test files. Alternatively, we can just define them as None Types.